app/src/main/java/com/example/nvdovin/weatherapp/adapter/RecyclerViewAdapter.java

```diff
@@ -0,0 +1,109 @@
+    private Context context;
+    private Typeface weatherFont;
+
+
+    public RecyclerViewAdapter(List<City> cityList, int tempScale, Context context) {
+        this.context = context;
+        this.cityList = new ArrayList<>();
+        this.cityList = cityList;
+        this.tempScale = tempScale;
+        weatherFont = Typeface.createFromAsset(this.context.getAssets(), FONTS_LOCATION);
+    }
+
+    @Override
+    public CustomViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(context).inflate(R.layout.forecast_recycler_row, parent, false);
+        return new CustomViewHolder(view);
+    }
+
+    @Override
+    public void onBindViewHolder(CustomViewHolder holder, int position) {
+        City city = cityList.get(position);
+        String weatherIconId = city.getWeatherDataList().get(0).getWeatherIcon();
+        holder.cityName.setText(city.getName());
+        holder.cityWeatherDescription.setText(city.getWeatherDataList().get(0).getWeatherDescription());
+        holder.cityTemperature.setText(String.valueOf(city.getWeatherDataList().get(0).getTemp().intValue() - tempScale));
+        holder.cityID = city.getId();
+        holder.itemView.setBackgroundResource(setWeatherIcon(weatherIconId, holder.weatherIcon));
+        holder.itemView.getBackground().setAlpha(TRANSPARENCY_ALPHA);
+    }
+
+
+    @Override
+    public int getItemCount() {
+        return cityList.size();
+    }
+
+
+    public class CustomViewHolder extends RecyclerView.ViewHolder {
+        TextView cityName, cityTemperature, cityWeatherDescription, weatherIcon;
+        Long cityID;
+
+        public CustomViewHolder(View itemView) {
+            super(itemView);
+            weatherIcon = (TextView) itemView.findViewById(R.id.forecast_recycler_weather_icon);
+            weatherIcon.setTypeface(weatherFont);
+            cityName = (TextView) itemView.findViewById(R.id.forecast_recycler_city_name);
+            cityTemperature = (TextView) itemView.findViewById(R.id.forecast_recycler_city_temperature);
+            cityWeatherDescription = (TextView) itemView.findViewById(R.id.forecast_recycler_city_weather);
+            itemView.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View v) {
+                    //To be edited
+                    Toast.makeText(v.getContext(), "You have selected " + cityName.getText() + " with the id - " + cityID, Toast.LENGTH_SHORT).show();
+
+                }
+            });
+        }
+    }
+    private int setWeatherIcon(String id, TextView iconTextView) {
+        WeatherCodesMap weatherCodesMap = new WeatherCodesMap();
+        int resID = weatherCodesMap.getBackgroundResById(id);
+        String icon = context.getString(weatherCodesMap.getIconByID(id));
+
+        if (Build.VERSION.SDK_INT >= MARSHMALLOW_VERSION)
+            iconTextView.setText(Html.fromHtml(icon, Html.FROM_HTML_MODE_LEGACY));
```

Comment:
needed for loading the icons from the webfont (fonts/weathericons-regular-webfont.ttf)